lib/Parse/ParseDecl.cpp

```diff
@@ -6748,6 +6748,17 @@ Parser::parseDeclVar(ParseDeclOptions Flags,
     // If we syntactically match the second decl-var production, with a
     // var-get-set clause, parse the var-get-set clause.
     if (Tok.is(tok::l_brace)) {
+
+      // Skip parsing the var-get-set clause if '{' is at start of line
+      // and next token is not 'didSet' or 'willSet'. Parsing as 'do'
+      // statement gives useful errors for missing 'do' before brace.
+      // See SR-14836.
+      Token NextToken = peekToken();
+      if (Tok.isAtStartOfLine() && !NextToken.isContextualKeyword("didSet") &&
+          !NextToken.isContextualKeyword("willSet")) {
```

Comment:
You should do this *only if* `PatternInit` is *not* null. For example:
```swift
var value: Int
{ return 1 }
```
This must be parsed as before. The brace is actually a start of the getter.